src/Controls/src/Build.Tasks/SetPropertiesVisitor.cs

```diff
@@ -417,13 +417,20 @@ static IEnumerable<Instruction> CompileBindingPath(ElementNode node, ILContext c
 				skipNode = GetParent(node);
 			}
 
+			bool xDataTypeIsInOuterScope = false;
 			while (n != null)
 			{
 				if (n != skipNode && n.Properties.TryGetValue(XmlName.xDataType, out dataTypeNode))
 				{
 					break;
 				}
 
+				if (n.XmlType.Name == nameof(Microsoft.Maui.Controls.DataTemplate)
+					&& n.XmlType.NamespaceUri == XamlParser.MauiUri)
```

Comment:
Are there any other _types_ of scopes we would need future changes for?

Wondering if any of these have an issue:

* `ResourceDictionary`
* `ControlTemplate`

In theory, you could have a `{Binding}` inside a `Style` in a `{ResourceDictionary}` and the type could be different based on the XAML using the `Style`? Maybe the existing warning already works for this case.

Reply:
I don't think this applies to ResourceDictionary (BTW we're not compiling these bindings at the moment: https://github.com/dotnet/maui/pull/22023), but it might apply to ControlTemplate 🤔 In general, this should be whenever there's a different BindingContext in the outer scope and in the inner scope. What do you think, @StephaneDelcroix?